cmd/podman/containers/exec.go

```diff
@@ -149,7 +149,11 @@ func exec(cmd *cobra.Command, args []string) error {
 		if err != nil {
 			return err
 		}
-		if err := execWait(nameOrID, seconds); err != nil {
+		err = execWait(nameOrID, seconds)
+		if errors.Is(err, define.ErrCanceled) {
+			return fmt.Errorf("timed out waiting for container: %s", nameOrID)
+		}
+		if err != nil {
```

Comment:
Might it make better sense to keep line 152 as it is, and put the `errors.Is()` inside the condition?

The way you have it now, it will mostly be doing `errors.Is(nil)`, which I'm sure is well defined somewhere, but why make a maintainer have to look up that behavior?
```go
    if err := execWait..; err != nil {
        if errors.Is(err, define.ErrCanceled) {  <----- *NOW* we are guaranteed that err != nil
```

Reply:
I agree.  The suggested error checking is a bit more idiomatic.